torch/csrc/jit/argument_spec.cpp

```diff
@@ -284,5 +284,29 @@ void ArgumentSpecCreator::specializeTypes(
   }
 }
 
+template <typename F, typename T>
```

Comment:
I don't get this, it seems like it is just being used to convert a ScalarType to an optional int8_t. Why not write the function that does directly that? The templates are hard to follow.

Reply:
Initially, I thought I'd have more enum classes that I'd need to convert, but then I forgot to clean it up. My bad.